src/game/GameWorldLoader.ts

```diff
@@ -0,0 +1,300 @@
+        weight: data.weight,
+        actions: {
+            left: actionFromData(data.actions.left, defaultParams, 'No'),
+            right: actionFromData(data.actions.right, defaultParams, 'Yes'),
+        },
+    }
+}
+
+/**
+ * Creates a map of event cars from GameWorlds event card data
+ *
+ * @param eventCardsData The event card data
+ * @param defaultParams Default params to use for a state reset
+ * @returns A map of event cards
+ */
+function eventCardsFromData(
+    eventCardsData: GameWorld['eventCards'],
+    defaultParams: Params,
+): { [x: string]: Card<Params> } {
+    const eventCards = Object.keys(eventCardsData).reduce<{
+        [x: string]: Card<Params>
+    }>((acc, key) => {
+        const data = eventCardsData[key]
+        acc[key] = cardFromData(data, defaultParams)
+        return acc
+    }, {})
+
+    for (const cardId in eventCards) {
+        const data = eventCardsData[cardId]
+        const eventCard = eventCards[cardId]
+
+        eventCard.actions.left.modifier = eventCardChain(
+            data.actions.left,
+            eventCards,
+            eventCard.actions.left.modifier,
+        )
+        eventCard.actions.right.modifier = eventCardChain(
+            data.actions.right,
+            eventCards,
+            eventCard.actions.right.modifier,
+        )
+    }
+    return eventCards
+}
+
+/**
+ * Completes an event card action modifier by adding trigger for next card
+ * in case a nextEventCardId is specified
+ *
+ * @param data The action data
+ * @param eventCards An event cards map
+ * @param modifier The current modifier without card trigger
+ * @returns A StateModifier with conditionally added next card trigger
+ */
+function eventCardChain(
+    data: EventCardActionData,
+    eventCards: { [x: string]: Card<Params> },
+    modifier: StateModifier<Params>,
+): StateModifier<Params> {
+    const targetCard =
+        data.nextEventCardId !== null ? eventCards[data.nextEventCardId] : null
+    return targetCard
+        ? (state) => ({
+              ...modifier(state),
+              card: targetCard,
+          })
+        : modifier
+}
+
+/**
+ * Creates a card action from CardActionData
+ *
+ * @param data
+ * @param defaultParams Default params to use when resetting state
+ * @param defaultDescription The default description in case it is missing in data
+ * @returns A card action with description and param modifier
+ */
+function actionFromData(
+    data: CardActionData,
+    defaultParams: Params,
+    defaultDescription: string,
+): CardAction<Params> {
+    return {
+        description: data.description ?? defaultDescription,
+        modifier: (state) => updateParams(state, data.modifiers, defaultParams),
+    }
+}
+
+/**
+ * Converts a WorldQuery to a ParamQuery
+ *
+ * @param query The WorldQuery
+ * @returns A ParamQuery representing the input WorldQuery
+ */
+function worldQueryToParamQuery(query: WorldQuery): ParamQuery {
+    return {
+        vars: query.state,
```

Comment:
We should really make another refactoring to replace `state` with `vars` in the `ContentTypes`. But this is another PR.

Then we can simplify the language when explaining the implementation + data layer to contributors. In this case, do we have any good reason to keep the `ParamQuery` separated from the `WorldQuery` type? Maybe we could even use `Param`s in the `ContentTypes`, to reduce the number of types and simplify learning?